src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -1476,6 +1474,18 @@ public Type getReturnType(String invocationId) {
 
             return handlers.get(0).getTypes();
         }
+
+        private void errorHandshake(Exception error) {
+            lock.lock();
+            try {
+                // If onError is called on a completed subject the global error handler is called
```

Comment:
Is there a way we could hook the global error handler with our tests and fail if it ever gets called? Would that have been able to catch this?

Reply:
you could use something like this:
https://github.com/uber/AutoDispose/blob/2.0.0/test-utils/src/main/java/autodispose2/test/RxErrorsRule.java

Reply:
Thanks for the pointer, adapted something similar for junit 5